components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/impl/UserIdApiServiceImpl.java

```diff
@@ -63,9 +71,30 @@ public Response terminateSessionBySessionId(String userId, String sessionId) {
     @Override
     public Response terminateSessionsByUserId(String userId) {
 
-        Util.validateUserId(SessionManagementServiceHolder.getRealmService(), userId,
-                ContextLoader.getTenantDomainFromContext());
-        sessionManagementService.terminateSessionsByUserId(userId);
-        return Response.noContent().build();
+        try {
+            UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+
+            if (userStoreManager == null) {
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+            }
+
+            String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            String adminUserName = userRealm.getRealmConfiguration().getAdminUserName();
+            String adminUserID = userStoreManager.getUserIDFromUserName(adminUserName);
+
+            if (!username.equals(adminUserName) && userId.equals(adminUserID)) {
+                return Response.status(Response.Status.FORBIDDEN).build();
```

Comment:
let's add a debug log here

Reply:
Fixed in commit abb7aa2b8853d2aac0c644c93951066787aa4c64